src/core/vectortile/qgsvectortileprovidermetadata.cpp

```diff
@@ -55,4 +55,35 @@ void QgsVectorTileProviderMetadata::saveConnection( const QgsAbstractProviderCon
   saveConnectionProtected( connection, name );
 }
 
+QgsProviderMetadata::ProviderCapabilities QgsVectorTileProviderMetadata::providerCapabilities() const
+{
+  return FileBasedUris;
+}
+
+QVariantMap QgsVectorTileProviderMetadata::decodeUri( const QString &uri ) const
+{
+  QgsDataSourceUri dsUri;
+  dsUri.setEncodedUri( uri );
+
+  QVariantMap uriComponents;
+  uriComponents.insert( QStringLiteral( "type" ), dsUri.param( QStringLiteral( "type" ) ) );
+  if ( uriComponents[ QStringLiteral( "type" ) ] == QStringLiteral( "mbtiles" ) )
+  {
+    uriComponents.insert( QStringLiteral( "path" ), dsUri.param( QStringLiteral( "url" ) ) );
+  }
+  else
+  {
+    uriComponents.insert( QStringLiteral( "url" ), dsUri.param( QStringLiteral( "url" ) ) );
+  }
+  return uriComponents;
```

Comment:
There's some other Uri components used which are missing here, breaking the lossless round trip. Eg zmin/zmax, serviceType, styleUrl, referer and the auth config.

Reply:
@nyalldawson , regarding auth config, I'm not sure we should handle it there (ATM, we don't in any of the providers which have implemented decodeUri/encodeUri).

Reply:
Actually they do: check afs, ams, hana, oracle, postgres.... 

If it's not handled then the lossless round trip contract is broken, which will break things

Reply:
oh I missed those. we re missing it for ogr and gdal then. I ll adjust.

Reply:
For the record I think it's also missing for the wms provider.